tactics/hints.ml

```diff
@@ -1662,11 +1662,46 @@ let pr_hint env sigma h = match h.obj with
   | Extern (_, tac) ->
     str "(*external*) " ++ Gentactic.print_glob env sigma ~level:(LevelLe 0) tac
 
+let replace_metas c env sigma clnv =
+  let open EConstr in
+  let metam = Clenv.clenv_meta_list clnv in
+  let sigma = ref sigma in
+  let mvMap : EConstr.t Int.Map.t ref = ref Int.Map.empty in
+  let rec upd c =
+    match kind !sigma c with
+    | Meta mv ->
+      begin match Unification.Meta.meta_name metam mv with
+      | Name na -> begin
+        try
+          Int.Map.find mv !mvMap
+        with Not_found ->
+          let (sigma',ev) = Evarutil.new_evar ~naming:(IntroIdentifier na) env !sigma c in
+          mvMap := Int.Map.add mv ev !mvMap;
+          sigma := sigma';
+          ev
+        end
+      | Anonymous -> c
+      end
+    | _ -> EConstr.map !sigma upd c
+  in
+  let c = upd c in
+  c,!sigma
+
+let pr_default_pattern env sigma (h0 : hint hint_ast) = match h0 with
+| Give_exact h ->
+  pr_leconstr_env env sigma h.hint_type
+| Res_pf h | ERes_pf h | Res_pf_THEN_trivial_fail h ->
+  let c, sigma = replace_metas (Clenv.clenv_type h.hint_clnv) env sigma h.hint_clnv in
+  pr_leconstr_env env sigma c
+| Unfold_nth _ | Extern _ ->
+  (* These hints cannot contain DefaultPattern *)
+  assert false
+
 let pr_id_hint env sigma (id, v) =
   let pr_pat p = match p.pat with
```

Comment:
@yannl35133 Do you know about the suggestion feature?  Go the "Files Changed" panel, select a range of lines (click/drag on the + on the left), click the icon (shown at bottom) then modify the text.  Note the subline highlighting of additions and deletions.

```suggestion
let pr_id_hint env sigma  = (* example *)
  let pr_pat p = match p.pat with
```

<img width="1597" height="151" alt="image" src="https://github.com/user-attachments/assets/b77b95bc-3fdb-47b9-9cb0-807262c64e96" />